ECIPs/ECIP-1056.md

```diff
@@ -0,0 +1,73 @@
+### ECIP-1056: Agharta EVM and Protocol Upgrades
+
+    ECIP: 1056
+    Title: Agharta EVM and Protocol Upgrades
+    Status: Draft
+    Type: Standards Track
+    Author: Isaac Ardis <isaac.a@etclabs.org>
+    Created: 2019-03-25
+
+### Simple Summary
+
+Enable the outstanding Ethereum Foundation _Constaninople_ and _Petersburg_ network protocol upgrades on the Ethereum
+Classic network in a hard-fork code-named _Agharta_ to enable maximum compatibility across these networks.
+
+### Abstract
+
+Add support for a subset of protocol-impacting changes introduced in the Ethereum Foundation (ETH) network via the
+_Constaninople_ and _Petersburg_ hardforks. The proposed changes for Ethereum Classic's _Agharta_ upgrade include:
+
+- Constantinople bitwise shifting instructions
+- Constantinople skinny `CREATE2` opcode
+- Constantinople `EXTCODEHASH` opcode
+
+This document proposes the following blocks at which to implement these changes in the Classic networks:
+
+- `1_500_000` on Kotti Classic PoA-testnet (early November 2019)
+- `5_200_000` on Morden Classic PoW-testnet (early November 2019)
+- `9_200_000` on Ethereum Classic PoW-mainnet (early December 2019)
+
+For more information on the opcodes and their respective EIPs and implementations, please see the _Specification_
+section of this document.
+
+### Motivation
+
+To enhance the Ethereum Virtual Machine's (EVM) capabilities, various opcodes shall be added to the Ethereum Classic
+networks, all of which have been in use on the Ethereum Foundation networks since early 2019.
+
+### Specification
+
+Technical specifications for each EIP can be found at those documents respectively:
+
+- [EIP 145](https://eips.ethereum.org/EIPS/eip-145) (Bitwise shifting instructions)
+- [EIP 1014](https://eips.ethereum.org/EIPS/eip-1014) (Skinny `CREATE2` opcode)
+- [EIP 1052](https://eips.ethereum.org/EIPS/eip-1052) (`EXTCODEHASH` opcode)
```

Comment:
We need a simple clarification on [EIP1052]. We have to specify somewhere (here?)
```
In case the account exists but is _the empty account_ 0 is pushed to the stack.
```

I believe this would fix the issue @sorpaas was warning about implementing "1052 without 161".

Reply:
should make clear that 1056 depends on 1054 :+1:

Reply:
ah, ok, but if we decide against 161 (since its so buggy), this will could prevent the issue